core/store/src/migrations.rs

```diff
@@ -186,3 +200,54 @@ pub fn migrate_31_to_32(storage: &crate::NodeStorage) -> anyhow::Result<()> {
     update.commit()?;
     Ok(())
 }
+
+fn get_row_count(store: &Store, col_name: &str) -> Option<i64> {
+    let statistics = store.get_store_statistics()?;
+    let col_stats = statistics.data.into_iter().find(|(col, _)| col == col_name)?.1;
+    for stat in col_stats {
+        match stat {
+            crate::db::StatsValue::Count(count) => return Some(count),
+            _ => {}
+        }
+    }
+    None
+}
+
+/// Migrates database from version 32 to 33.
+///
+/// This removes the TransactionResult column and moves it to TransactionResultForBlock.
+/// The new column removes the need for high-latency read-modify-write operations when committing
+/// new blocks.
+pub fn migrate_32_to_33(storage: &crate::NodeStorage) -> anyhow::Result<()> {
+    let store = storage.get_store(crate::Temperature::Hot);
+    let mut update = BatchedStoreUpdate::new(&store, 10_000_000);
+    let rows_total = get_row_count(&store, "col7");
+    let mut rows_migrated = 0;
+    for row in
+        store.iter_prefix_ser::<Vec<ExecutionOutcomeWithIdAndProof>>(DBCol::_TransactionResult, &[])
+    {
+        let (key, outcomes) = row?;
+        update.delete(DBCol::_TransactionResult, &key)?;
+        for outcome in outcomes {
+            update.set_ser(
+                DBCol::TransactionResultForBlock,
+                &get_outcome_id_block_hash(outcome.id(), &outcome.block_hash),
+                &ExecutionOutcomeWithProof {
+                    proof: outcome.proof,
+                    outcome: outcome.outcome_with_id.outcome,
+                },
+            )?;
+        }
+
+        rows_migrated += 1;
+        if rows_migrated % 100000 == 0 {
+            info!(
+                target: "migrations",
+                "Migrated {}/{} TransactionResult rows",
```

Comment:
Perhaps `Migrated {} of around {} TransactionResult rows`?

Reply:
The count actually returns None. So forget it - I made it print the number of total bytes written instead.